src/connection.ts

```diff
@@ -1967,21 +1981,52 @@ class Connection extends EventEmitter {
 
     connect(connectOpts, dns.lookup, signal).then((socket) => {
       process.nextTick(() => {
-        socket.on('error', (error) => { this.socketError(error); });
-        socket.on('close', () => { this.socketClose(); });
-        socket.on('end', () => { this.socketEnd(); });
-        socket.setKeepAlive(true, KEEP_ALIVE_INITIAL_DELAY);
+        const secureContext = tls.createSecureContext(this.secureContextOptions);
+        if (this.config.options.encrypt === 'strict') {
+          const encryptOptions = {
+            host: this.config.server,
+            socket: socket,
+            ALPNProtocols: ['tds/8.0'],
+            secureContext: secureContext,
+            servername: this.config.options.serverName ? this.config.options.serverName : this.config.server,
+          };
+          const encryptsocket = tls.connect(encryptOptions, () => {
```

Comment:
What will happen if someone connects using `strict` encryption to a SQL Server instance that does not support it?
How do we need to handle that, and how do we signal to the user that something is wrong? How do we help them understand _what_ is wrong?

Reply:
If user uses a strict to connect to a SQL server instance that does not support it, a socket error will be returned as : " Client network socket disconnected before secure TLS connection was established" 
Server side message: 
"The login packet used to open the connection is structurally invalid; the connection has been closed. Please contact the vendor of the client library. [CLIENT: 172.28.192.1]
"
 what do you think that we consume this message internally, and throw a custom error with a bit more information, so we can guide user through this